VRDR.Messaging/CodingResponseMessage.cs

```diff
@@ -311,6 +311,78 @@ public CauseOfDeathEntityAxisEntry(string deathCertificateText, string causeOfDe
         }
     }
 
+    /// <summary>
+    /// Helper class for building a List&lt;CauseOfDeathEntityAxis&gt; from a flat file. Groups codes
+    /// into one <c>CauseOfDeathEntityAxisEntry</c> per line and sorts codes by position.
+    /// </summary>
+    public class CauseOfDeathEntityAxisBuilder
+    {
+        private SortedDictionary<int, SortedDictionary<int, string>> codes;
+
+        /// <summary>
+        /// Construct a new empty instance.
+        /// </summary>
+        public CauseOfDeathEntityAxisBuilder()
+        {
+            codes = new SortedDictionary<int, SortedDictionary<int, string>>();
+        }
+
+        /// <summary>
+        /// Build a List&lt;CauseOfDeathEntityAxis&gt; from the currently contained set of codes.
+        /// </summary>
+        /// <returns>cause of death entity axis coding list</returns>
+        public List<CauseOfDeathEntityAxisEntry> ToCauseOfDeathEntityAxis()
+        {
+            var list = new List<CauseOfDeathEntityAxisEntry>();
+            foreach (KeyValuePair<int, SortedDictionary<int, string>> pair in codes)
+            {
+                var entry = new CauseOfDeathEntityAxisEntry("", pair.Key.ToString());
+                foreach (var code in pair.Value.Values)
+                {
+                    entry.AssignedCodes.Add(code);
+                }
+                list.Add(entry);
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Add a code to the list of codes that will be used to build a List&lt;CauseOfDeathEntityAxis&gt;.
+        /// Order of code addition is not significant, codes will be ordered by <c>line</c> and <c>position</c>
+        /// by the <c>ToCauseOfDeathEntityAxis</c> method.
+        /// </summary>
+        /// <param name="line">The line number from the death certificate:
+        /// <list type="number">
+        /// <item>Part I. Line a</item>
+        /// <item>Part I. Line b</item>
+        /// <item>Part I. Line c</item>
+        /// <item>Part I. Line d</item>
+        /// <item>Part I. Line e</item>
+        /// <item>Part II</item>
+        /// </list>
+        /// </param>
+        /// <param name="position">Sequence within line</param>
+        /// <param name="code">ICD code</param>
+        /// <exception cref="System.ArgumentException">Thrown if <c>line</c> or <c>position</c> is not a number</exception>
+        public void Add(string line, string position, string code)
```

Comment:
Maybe `lineNumber` would be more explicit than `line`? (And then `lineNumberValue` below)

Reply:
OK.